agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1015,6 +1019,10 @@ func (dg *dockerGoClient) handleContainerEvents(ctx context.Context,
 		}
 
 		metadata := dg.containerMetadata(ctx, containerID)
+		// In case when we received a container die event but was not able to inspect the container (e.g. due to timeout),
+		// we will use the exit code from the event, so that the exit code of the container is still reported and
+		// available for customer to see from describing task.
```

Comment:
if exit code is set here, do we still need the inspect later?

Reply:
yeah, because we need to get the exitAt timestamp https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/container/container.go#L310. this is shown in the metadata, and is only available by inspecting the container (see https://github.com/moby/moby/blob/master/daemon/monitor.go#L48, the exit code is stored in event but exitAt is not). otherwise i would have remove the inspect :disappointed:

Reply:
got it. if inspect fails, do we still show the error in task stopped reason even when container exits fine?

Reply:
yes we will show the error in the reason field. see the Testing section in pr description for an example